api/queryio/writer.go

```diff
@@ -33,9 +35,17 @@ func NewWriter(w io.WriteCloser, format string, flusher http.Flusher) (*Writer,
 	var err error
 	switch format {
 	case "zng":
-		d.writer = NewZNGWriter(w)
+		if ctrl {
+			d.writer = NewZNGWriter(w)
+		} else {
+			d.writer = zngio.NewWriter(w)
+		}
 	case "zjson":
-		d.writer = NewZJSONWriter(w)
+		if ctrl {
+			d.writer = NewZJSONWriter(w)
+		} else {
+			d.writer = zjsonio.NewWriter(w)
```

Comment:
I think what's going on here would be substantially clearer if we added a `ctrl bool` field to `Writer` and then checked it in `WriteControl`.

Reply:
I can dig it 👍🏻